website/docs/r/api_management_email_template.html.markdown

```diff
@@ -0,0 +1,104 @@
+}
+
+resource "azurerm_api_management" "example" {
+  name                = "example-apim"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  publisher_name      = "My Company"
+  publisher_email     = "company@terraform.io"
+
+  sku_name = "Developer_1"
+}
+
+resource "azurerm_api_management_email_template" "example" {
+  template_name       = "ConfirmSignUpIdentityDefault"
+  resource_group_name = azurerm_resource_group.example.resource_group_name
+  api_management_name = azurerm_api_management.example.name
+  subject             = "Customized confirmation email for your new $OrganizationName API account"
+  body                = <<EOF
+<!DOCTYPE html >
+<html>
+<head>
+  <meta charset="UTF-8" />
+  <title>Customized Letter Title</title>
+</head>
+<body>
+  <p style="font-size:12pt;font-family:'Segoe UI'">Dear $DevFirstName $DevLastName,</p>
+</body>
+</html>
+EOF
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `template_name` - (Required) The name of the Email Template. Possible values are `AccountClosedDeveloper`, `ApplicationApprovedNotificationMessage`, `ConfirmSignUpIdentityDefault`, `EmailChangeIdentityDefault`, `InviteUserNotificationMessage`, `NewCommentNotificationMessage`, `NewDeveloperNotificationMessage`, `NewIssueNotificationMessage`, `PasswordResetByAdminNotificationMessage`, `PasswordResetIdentityDefault`, `PurchaseDeveloperNotificationMessage`, `QuotaLimitApproachingDeveloperNotificationMessage`, `RejectDeveloperNotificationMessage`, `RequestDeveloperNotificationMessage`. Changing this forces a new API Management Email Template to be created.
+
+* `api_management_name` - (Required) The name of the API Management Service in which the Email Template should exist. Changing this forces a new API Management Email Template to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group where the API Management Email Template should exist. Changing this forces a new API Management Email Template to be created.
+
+* `subject` - (Required) The subject of the Email.
+
+* `body` - (Required) The body of the Email. It has to be a well-formed HTML document.
+
+-> **NOTE:** In `subject` and `body` predefined parameters can be used. Possible values to use are `$DevFirstName`, `$DevLastName`, `$OrganizationName`, `$DevPortalUrl`, `$ConfirmUrl`, `$DevPortalHost`, `$ConfirmQuery`.
+
+## Attributes Reference
+
+In addition to the Arguments listed above - the following Attributes are exported:
+
+* `id` - The ID of the API Management Email Template.
+
+* `title` - The title of the Email Template.
+
+* `description` - The description of the Email Template.
+
+* `parameters` - A `parameters` block as defined below.
+
+---
+
+A `parameters` block exports the following:
+
+* `name` - The name of this parameter.
+
+* `title` - The title of this parameter.
+
+* `description` - The description of this parameter.
```

Comment:
are these parameters the ones listed above? if so, maybe we don't need to export these if they're already known?

Reply:
Yes, those are the `$DevFirstName` parameters. After you brought my attention to this, I double checked and realized that the available parameters depend on the template. So I updated the *Note* above.

My question is how to proceed with those parameters?
It is not plausible to reference the parameters in e.g. the `subject` in the tf resource because you would have a self-reference. Also you would need to know the parameter in advance.
We could just not export them (remove them from the resource schema definition) and put a reference in the *Note* that one can look them up in the Azure Portal - Notification Template section of the ApiManagement Service instance or Management REST API.
Or export them and hint that they also can be seen in the tf state?
Or even list all available parameters for each of the possible templates?

What do you think? :)

Reply:
I just pushed my proposal: I removed the parameters and just mentioned in the *Note* of the documentation where one can find the available values.